### PR TITLE
Add support for new application command option types

### PR DESCRIPTION
**Pull request by IDoEverything**
# Summary
Adds support for the `Mentionable` and `Number` application command option types.

# Details
Mentionable can be a user or a role, and apparently you're supposed to check which one it is from `Resolved`. Number allows for float/double values.

### CODE REVIEW

**Review by Lulalaby — APPROVED** [review]

File: DSharpPlus/Enums/Interaction/ApplicationCommandOptionType.cs [hunk 1]
```diff
@@ -66,6 +66,16 @@ public enum ApplicationCommandOptionType
...
-        Role
+        Role,
+
+        /// <summary>
+        /// Whether this parameter is a mentionable (role or user).
+        /// </summary>
+        Mentionable,
+
+        /// <summary>
+        /// Whether this paramter is a double.
```
Comment (Kiritsu) [inline DSharpPlus/Enums/Interaction/ApplicationCommandOptionType.cs:77]: parameter*

**Review by Kiritsu — CHANGES_REQUESTED** [review]

**Review by Kiritsu — APPROVED** [review]

**Comment by Neuheit** [comment]
Will be merged when https://github.com/discord/discord-api-docs/pull/3455 is merged.

**Comment by IDoEverything** [comment]
It's been merged

**Comment by Lulalaby** [comment]
yes. @Neuheit 

@IDoEverything Atm I'm implementing this into slashcommands, but I've a few problems.
I'll make a pr if mine works.

**Comment by Lulalaby** [comment]
@IDoEverything any idea?

Value: 0.2
Slash/Error: Input string was not in a correct format. | CN: volume | IID: 869308250892402709

DiscordInteractionDataOption
```csharp
        /// <summary>
        /// Gets the value of this interaction parameter.
        /// <para>This can be cast to a <see langword="long"/>, <see langword="bool"></see>, <see langword="string"></see>, <see langword="double"></see> or <see langword="ulong"/> depending on the <see cref="Type"/></para>
        /// </summary>
        [JsonIgnore]
        public object Value
        {
            get
            {
                Console.WriteLine("Value: " + this._value);
                return this.Type switch
                {
                    ApplicationCommandOptionType.Boolean => bool.Parse(this._value),
                    ApplicationCommandOptionType.Integer => long.Parse(this._value),
                    ApplicationCommandOptionType.String => this._value,
                    ApplicationCommandOptionType.Channel => ulong.Parse(this._value),
                    ApplicationCommandOptionType.User => ulong.Parse(this._value),
                    ApplicationCommandOptionType.Role => ulong.Parse(this._value),
                    ApplicationCommandOptionType.Mentionable => ulong.Parse(this._value),
                    ApplicationCommandOptionType.Number => double.Parse(this._value, NumberStyles.AllowDecimalPoint),
                    _ => this._value,
                };
            }
        }
```

But if I remove NumberStyles.AllowDecimalPoint
It resolves to double 2

----

Value: 0.2
Type: Nyaw.SlashCommands.Owner+VolumeChoiceProvider, Var: 2

This i get when i use the following code

DiscordInteractionDataOption
```csharp
        /// <summary>
        /// Gets the value of this interaction parameter.
        /// <para>This can be cast to a <see langword="long"/>, <see langword="bool"></see>, <see langword="string"></see>, <see langword="double"></see> or <see langword="ulong"/> depending on the <see cref="Type"/></para>
        /// </summary>
        [JsonIgnore]
        public object Value
        {
            get
            {
                Console.WriteLine("Value: " + this._value);
                return this.Type switch
                {
                    ApplicationCommandOptionType.Boolean => bool.Parse(this._value),
                    ApplicationCommandOptionType.Integer => long.Parse(this._value),
                    ApplicationCommandOptionType.String => this._value,
                    ApplicationCommandOptionType.Channel => ulong.Parse(this._value),
                    ApplicationCommandOptionType.User => ulong.Parse(this._value),
                    ApplicationCommandOptionType.Role => ulong.Parse(this._value),
                    ApplicationCommandOptionType.Mentionable => ulong.Parse(this._value),
                    ApplicationCommandOptionType.Number => Convert.ToDouble(this._value),
                    _ => this._value,
                };
            }
        }
```

```csharp

        [SlashCommand("volume", "Modifies the volume")]
        public static async Task ModifyVolumeAsync(InteractionContext ctx, [ChoiceProvider(typeof(VolumeChoiceProvider))][Option("volume", "Volume")] double volume)
        {
            Console.WriteLine($"Type: {typeof(VolumeChoiceProvider)}, Var: {volume}");
            await ctx.CreateResponseAsync(InteractionResponseType.DeferredChannelMessageWithSource, new DiscordInteractionResponseBuilder().WithContent("Changing Volume"));
            VoiceNextExtension vnext = ctx.Client.GetVoiceNext();
            VoiceNextConnection vnc = vnext.GetConnection(ctx.Guild);
            if (vnc == null)
                return;
            VoiceTransmitSink ts = vnc.GetTransmitSink();
            //ts.VolumeModifier = volume;
            await ctx.EditResponseAsync(new DiscordWebhookBuilder().WithContent($"Set volume to {ts.VolumeModifier}"));
        }
```

till `Console.WriteLine("Value: " + this._value);` it's correct.
after that, it converts wrong

**Comment by IDoEverything** [comment]
image
works for me

**Comment by Lulalaby** [comment]
crazy

**Review by Neuheit — APPROVED** [review]

**Pull request merged**